camp_fin/management/commands/aggregate_data.py

```diff
@@ -38,6 +38,31 @@ def makeTransactionAggregates(self, recreate_views):
                     )
                 )
 
+            continuous_interval_query = """
+                CREATE MATERIALIZED VIEW {transaction_type}_by_{interval} AS (
+                    WITH {interval}_summaries AS (
+                        {summary_query}
+                    )
+                    SELECT
+                        continuous_interval.entity_id,
+                        continuous_interval.{interval},
+                        COALESCE({interval}_summaries.amount, 0) AS amount
+                    FROM (
+                        SELECT DISTINCT
+                            entity_id,
+                            generate_series(
+                                MIN({interval}) OVER (PARTITION BY entity_id),
+                                MAX({interval}) OVER (PARTITION BY entity_id),
+                                '1 {interval}'
+                            ) AS {interval}
+                        FROM {interval}_summaries
+                    ) continuous_interval
+                    LEFT JOIN {interval}_summaries
+                    USING (entity_id, {interval})
+                    ORDER BY entity_id, {interval}
+                )
+            """
```

Comment:
If we have transaction info for a candidate in Jan, March, and April, then we can infer from the absence of transactions in Feb that the amount should be 0. This query creates a continuous sequence of aggregates for each entity, beginning with the first date we have a transaction for and ending with the last date we have a transaction for.

Reply:
This is useful in particular for the chart. Given sparse data, a step chart will draw a line from point A to C, suggesting a potentially misleading value for B (which gets worse the more values are missing). This registers months without transactions appropriately as 0.